src/python/WMCore/WorkQueue/WMBSHelper.py

```diff
@@ -348,18 +349,14 @@ def addMCFakeFile(self):
             if self.mask and self.mask.get(key) is None:
                 msg = 'Invalid value "%s" for %s' % (self.mask.get(key), key)
                 raise WorkQueueWMBSException(msg)
+
         locations = set()
-        for site in self.getLocations.execute(conn=self.getDBConn(),
-                                              transaction=self.existingTransaction()):
-            try:
-                siteInfo = self.getLocationInfo.execute(site, conn=self.getDBConn(),
-                                                        transaction=self.existingTransaction())
-                if not siteInfo:
-                    self.logger.info('Skipping MonteCarlo injection to site "%s" as unknown to wmbs' % site)
-                    continue
-                locations.add(siteInfo[0]['pnn'])
-            except Exception as ex:
-                self.logger.error('Error getting storage element for "%s": %s' % (site, str(ex)))
+        siteInfo = self.getLocationInfo.execute(conn=self.getDBConn(),
```

Comment:
@amaltaro I have only one question. Why did you take the siteInfo query out of the 'try: - except:' block?

Reply:
It should be caught upstream, without breaking the whole cycle, but only this one element.
I also removed it, because as you can see in the old line 352, it wasn't covered by any try/except block as well.